gensim/similarity_learning/preprocessing/list_generator.py

```diff
@@ -0,0 +1,176 @@
+import pandas as pd
+import numpy as np
+from collections import Counter
+import logging
+import re
+
+logger = logging.getLogger(__name__)
+
+
+class ListGenerator:
+    """Class to generate the data for validation and testing in a list format.
+    Givem a dataset, this object will provide lists to be ranked by a trained model
+
+    List format is given through the `get_list_data` funciton which returns a dict
+    with the list format:
+    {"X1": X1, "X2": X2, "y": y, "doc_lengths": doc_lengths}
+
+    Example Usage:
+    -------------
+    list_generator = JListGenerator(test_file_path, text_maxlen=200)

```

Comment:
`JListGenerator`? Maybe `ListGenerator`?

Reply:
These were artifacts of my tweaking which I missed while cleaning.